qutip/fileio.py

```diff
@@ -217,15 +216,18 @@ def qsave(data, name='qutip_data'):
     ----------
     data : instance/array_like
         Input Python object to be stored.
-    filename : str
+    filename : str or pathlib.Path
         Name of output data file.
 
     """
     # open the file for writing
-    fileObject = open(name + '.qu', 'wb')
-    # this writes the object a to the file named 'filename.qu'
-    pickle.dump(data, fileObject)
-    fileObject.close()
+    file = Path(name)
+    if file.suffix != '.qu':
+        file.with_suffix('.qu')
```

Comment:
The `file.with_suffix(...)` doesn't replace the suffix of `file`. It returns a new path object which is then discarded because it isn't assigned to anything.